### DISCUSSION TITLE
Configuration to override parameter name if type is array (or something else)

### DISCUSSION THREAD

**Discussion by egreb** [orval-labs/orval#163] [Q&A] [answered]
Hello. 

I am using orval with a dotnet core backend, and I am trying to bind a array of numbers into a class property.

```c#
public class MyRequest 
{
 public int[] ArrayOfNumbers {get;set}
}
```

Then the controllere usage looks like something like this:
```c#
public async Task<IActionResult> Get([FromQuery] MyRequest query)
{
}
```

The problem is that dotnet adds `[]` at the end of `ArrayOfNumbers` in the request query, I can fix this by adding:
```c#
public class MyRequest 
{
 [FromQuery(Name = "ResponsibleIds[]")]
 public int[] ArrayOfNumbers {get;set}
}
```

The problem now is that the generated code from orval wants me to use a property named "ArrayOfNumbers[]", which then will be considered as "ArrayOfNumbers[][]" in the backend. 

Is there any way I can fix this with configuring orval to use `ArrayOfNumbers` and not `ArrayOfNumbers[]` despite the ´[FromQuery]` binding?

This is a bit confusing, and my explanation could probably be a bit better sorry.

**Comment by egreb** [comment, marked as answer]
I solved this by overriding the axios config to remove [] from the query string.

**Reply by LanVukusic** [reply]
Could you please share the code that fixed that for you.  
I'm having a similar problem with `react-query`.

**Reply by LanVukusic** [reply]
After some research i found out that `axios` in the end serializes the request even if you use any other client (`react-query` in my example).

To solve this you have to override the `request` part.

```ts
  const { data, isLoading } = useGetQueryNameExample(
    {
      param1: "something",
      param2: "something-else"
    },
    {
      request: {
        // To make java happy
        paramsSerializer: { indexes: null },
      },
    },
  );
```
More on it can be read at [axios Request Config here](https://axios-http.com/docs/req_config).
